internal/encoder/encode_race.go

```
@@ -28,7 +28,15 @@ import (
 
 
 func helpDetectDataRace(val interface{}) {
-    _, _ = json.Marshal(val)
+    var out []byte
+    defer func() {
+        if v := recover(); v != nil {
```

just let it panic here? if we truncate the panic info, more details maybe lost